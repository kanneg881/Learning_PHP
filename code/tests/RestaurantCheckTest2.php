<?php
include 'restaurant-check.php';

use PHPUnit\Framework\TestCase;

class RestaurantCheckTest extends TestCase
{

    /**
     * 測試稅率和小費
     */
    public function testWithTaxAndTip(): void
    {
        /** @var int $meal 餐費 */
        $meal = 100;
        /** @var int $tax 稅率 */
        $tax = 10;
        /** @var int $tip 小費 */
        $tip = 20;
        /** @var float $result 總額 */
        $result = restaurantCheck($meal, $tax, $tip);
        $this->assertEquals(130, $result);
    }

    /**
     * 測試稅率
     */
    public function testWithNoTip(): void
    {
        /** @var int $meal 餐費 */
        $meal = 100;
        /** @var int $tax 稅率 */
        $tax = 10;
        /** @var int $tip 小費 */
        $tip = 0;
        /** @var float $result 總額 */
        $result = restaurantCheck($meal, $tax, $tip);
        $this->assertEquals(120, $result);
    }
}
