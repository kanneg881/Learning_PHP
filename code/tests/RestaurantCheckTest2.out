PHPUnit 7.2.4 by Sebastian Bergmann and contributors.

.F                                                                  2 / 2 (100%)

Time: 129 ms, Memory: 13.50MB

There was 1 failure:

1) RestaurantCheckTest::testWithNoTip
Failed asserting that 110.0 matches expected 120.

RestaurantCheckTest.php:38

FAILURES!
Tests: 2, Assertions: 2, Failures: 1.
